test/common/core/coretype.spec.js

```diff
@@ -1161,11 +1161,30 @@ describe('coretype', function () {
                     paths.push(core.getPath(instances[0]));
                     paths.push(core.getPath(instances[1]));
                     expect(paths).to.have.members(['/template', '/template/child']);
-                    if(--neededChecks === 0){
+                    if (--neededChecks === 0) {
                         done();
                     }
                 }, core.loadInstances(core.getBase(newChild)));
             }, core.loadByPath(newRoot, '/template/child'));
         }, core.loadRoot(core.getHash(root)));
     });
+
+    it('should remove atr and reg field of an instance during persist', function (done) {
```

Comment:
That makes sense - it looks like the check on line 180 (checking if the attr/reg is empty) was simply the problem for this issue